src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -470,10 +472,17 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             # make sure we only download sub-folders and `diffusers` filenames
             folder_names = [k for k in config_dict.keys() if not k.startswith("_")]
             allow_patterns = [os.path.join(k, "*") for k in folder_names]
-            allow_patterns += [WEIGHTS_NAME, SCHEDULER_CONFIG_NAME, CONFIG_NAME, ONNX_WEIGHTS_NAME, cls.config_name]
+            allow_patterns += [
+                WEIGHTS_NAME,
+                FLAX_WEIGHTS_NAME,
```

Comment:
Actually I think we should only add `FLAX_WEIGHTS_NAME` and remove the `"*.msgpack"` patterns if `from_flax` i- True:
```suggestion
```